src/Traits/PdfAttachmentTrait.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/*
+ * This file is part of the 'fpdf' package.
+ *
+ * For the license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @author bibi.nu <bibi@bibi.nu>
+ */
+
+declare(strict_types=1);
+
+namespace fpdf\Traits;
+
+use fpdf\PdfDocument;
+
+/**
+ * Trait to add file attachment support.
+ *
+ * The code is inspired from FPDF script
+ * <a href="http://www.fpdf.org/en/script/script95.php" target="_blank">Attachment</a>.
+ *
+ * @author Andreas Müller <hello@devmount.com>
+ *
+ * @phpstan-type PdfAttachedFileType = array{
+ *      file: string,
+ *      name: string,
+ *      desc: string,
+ *      n:    int}
+ *
+ * @phpstan-require-extends PdfDocument
+ */
+trait PdfAttachmentTrait
+{
+    /**
+     * The attached files.
+     *
+     * @phpstan-var PdfAttachedFileType[]
+     */
+    private array $files = [];
+
+    /**
+     * The files object number.
+     */
+    private int $nFiles = -1;
+
+    /**
+     * Flag to open the attachment pane in a PDF reader per default.
+     */
+    private bool $openAttachmentPane = false;
+
+    /**
+     * Attaches a given file to the PDF document.
+     *
+     * @param string $file path to the file to be attached
+     * @param string $name Optional alternative file name to be used for the attachment. The default value is taken
+     *                     from $file.
+     * @param string $desc optional description for the file contents
+     */
+    public function attach(string $file, string $name = '', string $desc = ''): static
+    {
+        if ('' === $name) {
```

Comment:
It better to use the `basename` function like this:

```php
if ('' === $name) {
    $name = \basename($file);
}
```

Reply:
Good catch, makes sense!